### DISCUSSION TITLE
Unpoly and Toast/Flash messages

### DISCUSSION THREAD

**Discussion by apollo13** [unpoly/unpoly#265] [Q&A]
Now that my python unpoly library is working, I wanted to show flash/toast messages to the user (basically when a layer is accepted I want to show "user xyz created successfully" somewhere). In Django one usually displays those below the header like this:
![image](https://user-images.githubusercontent.com/81547/130793743-392fd4f2-aadc-4002-b910-60fd1212dbc4.png)

Now with unpoly I do not really know (or rather don't want to care about) how many layers are open and if the parent layer could show that notification nicely. So I thought instead I could serialize the message and send it down as event and display it with something like like https://apvarun.github.io/toastify-js/# to show the messages.

Do you have any other ideas on how to approach this? This seems a rather common pattern. One example where it could make sense to include such a flash/toast message would be on https://demo.unpoly.com/companies for instance. When opening a company here and deleting it, there is no feedback to the user. Interestingly enough the list page doesn't seem to reload either after deleting a company -- so one only sees it after a hard refresh (bug, or is my browser hating me).

**Comment by flihub** [comment]
```html
<div id='flash-messages' up-hungry>
  <div>Message</div>
  <div>Message 2</div>
</div>
```
https://unpoly.com/up-hungry

loop and show messages when fragment:inserted
https://unpoly.com/up:fragment:inserted

**Reply by apollo13** [reply]
Mhm, the messages will be part of the dismissed/accepted layer, I am not sure if those would bubble up -- let me try that and report back.

**Reply by apollo13** [reply]
I do not see how this works. I have prepared a glitch here: https://glitch.com/edit/#!/colossal-familiar-industry -- where would I loop to get the messages when rendered in the overlay before it is accepted? Hook into `up:fragment:loaded` and copy them out?

**Comment by flihub** [comment]
up-hungry not updated if new layer and not in up-target .. maybe bug:

up-accept-location not trigger when up:fragment:inserted

removed up-accept-location
https://glitch.com/edit/#!/actually-alkaline-airship?path=index.html%3A21%3A10

**Reply by apollo13** [reply]
> up-hungry not updated if new layer and not in up-target .. maybe bug:

I just tried it, up-hungry works as expected. But the up-hungry will only affect the content in the new layer. There doesn't seem to be a way to bubble this up so to say.
